routing.rst

```diff
@@ -794,6 +794,52 @@ parameter:
     To give a ``null`` default value to any parameter, add nothing after the
     ``?`` character (e.g. ``/blog/{page?}``).
 
+Priority Parameter
+~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 5.1
+
+    The ``priority`` parameter was introduced in Symfony 5.1
+
+When defining a greedy pattern that matches many routes, this may be
+at the beginning of your routing collection and prevents any route defined after to be matched.
+A ``priority`` optional parameter is available in order to let you choose the order of your routes,
+and it is only available when using annotations to define routes.
+
+.. code-block:: php-annotations
+
+    // src/Controller/BlogController.php
+    namespace App\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\Routing\Annotation\Route;
+
+    class BlogController extends AbstractController
+    {
+        // ...
+
+        /**
+        * @Route("/{greedyParam}", name="greedy_route", priority=-2048)
+        */
+        public function greed(string $greedyParam)
+        {
+            // ...
+        }
+
+        /**
+        * @Route("/blog/{slug}", name="blog_show", priority=1)
+        */
+        public function show(string $slug)
+        {
+            // ...
+        }
+        // This route will be matched thanks to a higher priority which will place it ahead in the route collection
```

Comment:
Same as above, we could move the comment in the doc block and something like:
`Using the default 0 priority is higher than before, so the route will be tested in the same order as it is defined in the collection`

Reply:
On this point i would prefer `so the route will be matched firstly`, in order to insist on matching order, considering routes are tested in the collection's order in any case, do you agree ?

Reply:
I understand your point. In fact this `greedy` route is not really consistent with the `BlogController` class, ans the `-2048` priority not very readable. Let's use the "blog_show" route as greedy instead with a default priority. And add below a "blog_edit" route like "/blog/edit/{slug}" with a comment saying that it could not be matched unless defined a higher priority (anything above the 0 default). What do you think?